code/game/objects/structures/chisel_statue.dm

```diff
@@ -0,0 +1,261 @@
+	break_sculpting()
+
+/obj/item/chisel/proc/set_block(obj/structure/carving_block/B,mob/living/user)
+	prepared_block = B
+	tracked_user = user
+	RegisterSignal(tracked_user,COMSIG_MOVABLE_MOVED,.proc/break_sculpting)
+	to_chat(user,span_notice("You prepare to work on [B]."))
+
+/obj/item/chisel/dropped(mob/user, silent)
+	. = ..()
+	break_sculpting()
+
+/obj/item/chisel/proc/break_sculpting()
+	SIGNAL_HANDLER
+	sculpting = FALSE
+	if(prepared_block && prepared_block.completion == 0)
+		prepared_block.reset_target()
+	prepared_block = null
+	if(tracked_user)
+		UnregisterSignal(tracked_user,COMSIG_MOVABLE_MOVED)
+		tracked_user = null
+
+
+/obj/item/chisel/Destroy()
+	prepared_block = null
+	tracked_user = null
+	return ..()
+
+/obj/item/statue_kit
+	name = "sculpting block construction kit"
+	desc = "To make a sculpting block, you need to hit this with material sheets."
+
+/obj/item/statue_kit/attackby(obj/item/O, mob/user)
+	if(!istype(O, /obj/item/stack/material))
+		return ..()
+	var/obj/item/stack/material/M = O
+	if(M.use(2))
+		var/obj/structure/carving_block/CB = new /obj/structure/carving_block(get_turf(src))
+		CB.matter = list(M.material = 2)
+	else
+		to_chat(user, SPAN_NOTICE("You need 2 sheets of material to make a block."))
+
+/obj/structure/carving_block
+	name = "block"
+	desc = "ready for sculpting."
+	icon = 'icons/obj/statue.dmi'
+	icon_state = "block"
+	density = TRUE
+
+	/// The thing it will look like - Unmodified resulting statue appearance
+	var/current_target
+	/// Currently chosen preset statue type
+	var/current_preset_type
+	//Table of required materials for each non-abstract statue type
+	var/static/list/statue_costs
+	/// statue completion from 0 to 1.0
+	var/completion = 0
+	/// Greyscaled target with cutout filter
+	var/mutable_appearance/target_appearance_with_filters
+	/// HSV color filters parameters
+	var/static/list/greyscale_with_value_bump = list(0,0,0, 0,0,0, 0,0,1, 0,0,-0.05)
+
+/obj/structure/carving_block/Destroy()
+	current_target = null
+	target_appearance_with_filters = null
+	return ..()
+
+/obj/structure/carving_block/proc/set_target(atom/movable/target,mob/living/user)
+	if(!is_viable_target(target))
+		to_chat(user,"You won't be able to carve that.")
+		return
+	if(istype(target,/obj/structure/statue/custom))
+		var/obj/structure/statue/custom/original = target
+		current_target = original.content_ma
+	else
+		current_target = target.appearance
+	var/mutable_appearance/ma = current_target
+	to_chat(user,span_notice("You decide to sculpt [src] into [ma.name]."))
+
+/obj/structure/carving_block/proc/reset_target()
+	current_target = null
+	current_preset_type = null
+	target_appearance_with_filters = null
+
+/obj/structure/carving_block/update_icon()
+	..()
+	var/list/new_overlays = update_overlays()
+		cut_overlay(managed_overlays)
+	if(length(new_overlays))
+		add_overlay(new_overlays)
+
+/obj/structure/carving_block/proc/update_overlays()
+	. = ..()
```

Comment:
This is a new proc, ..() will not do anything

Reply:
![image](https://user-images.githubusercontent.com/22850904/162572841-6536d6b2-a6bb-444b-bccf-dea98422284e.png)